lib/Service.php

```diff
@@ -115,6 +115,11 @@ public function parse($input, string $contextUri = null, string &$rootElementNam
             // Unfortunately the XMLReader doesn't support streams. When it
             // does, we can optimize this.
             $input = (string) stream_get_contents($input);
+
+            // If input is an empty string, then its safe to throw exception
+            if ('' === $input) {
+                throw new ParseException('The input element to parse is empty. Do not attempt to parse');
```

Comment:
not sure I get the meaning of this error message? what do you try to tell the caller?